include/aikido/planner/vectorfield/VectorFieldIntegrator.hpp

```diff
@@ -0,0 +1,136 @@
+#include <aikido/planner/vectorfield/VectorFieldPlannerStatus.hpp>
+#include <aikido/planner/vectorfield/VectorFieldUtil.hpp>
+#include <aikido/statespace/dart/MetaSkeletonStateSpace.hpp>
+#include <aikido/trajectory/Spline.hpp>
+
+namespace aikido {
+namespace planner {
+namespace vectorfield {
+
+/// VectorField Planner generates a trajectory by following a vector field
+/// defined in joint space.
+/// A planned trajectroy ends either by a predefined termination criterion or a
+/// integral time.
+///
+/// This class defines two callback functions for a integrator.
+/// step() provides joint velocities the vector field planner should follow,
+/// check() is called after each integration step to check planner status.
+class VectorFieldIntegrator
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] vectorField Vector field in configuration space.
+  /// \param[in] constraint Constraint to be satisfied.
+  /// \param[in] initialStepSize Initial step size in integation.
+  VectorFieldIntegrator(
+      const VectorFieldPtr vectorField,
+      const aikido::constraint::TestablePtr collisionFreeConstraint,
+      double initialStepSize);
+
+  /// Generate a trajectory following the vector field along given time.
+  ///
+  /// \param[in] integrationTimeInterval Position in configuration space.
+  /// \param[in] timelimit Timelimit for integration calculation.
+  /// \param[out] planningResult information about success or failure.
+  /// \return A trajectory following the vector field.
+  std::unique_ptr<aikido::trajectory::Spline> followVectorField(
+      const aikido::statespace::StateSpace::State* startState,
+      std::chrono::duration<double> timelimit,
+      planner::PlanningResult& planningResult);
+
+  /// Convert a sequence of waypoint and time pairs into a trajectory.
+  ///
+  /// \param[in] A seqeunce of waypoint and time pairs.
+  /// \param[in] stateSpace State space of output trajectory
+  /// \return A trajectory
+  std::unique_ptr<aikido::trajectory::Spline> convertToSpline(
+      const std::vector<Knot>& knots,
+      const aikido::statespace::StateSpacePtr stateSpace);
+
+  /// Get initial step size.
+  ///
+  /// \return Initial step size.
+  double getInitialStepSize();
+
+  /// Get resolution of constraint checking.
+  ///
+  /// \return Resolution of constraint checking.
+  double getConstraintCheckResolution();
+
+  /// Set resolution of constraint checking.
+  ///
+  /// /param[in] Resolution used in constraint checking.
+  void setConstraintCheckResolution(double resolution);
+
+protected:
+  /// Evaluate whether a trajectory is eligible
+  ///
+  ///
+  virtual bool evaluateTrajectory(
+      const aikido::trajectory::Trajectory* trajectory,
+      const aikido::constraint::TestablePtr collisionFreeConstraint,
+      double evalStepSize)
+      = 0;
+
+  virtual bool convertStateToPositions(
+      const aikido::statespace::StateSpace::State* state,
+      Eigen::VectorXd& positions)
+      = 0;
```

Comment:
It's possible to implement this function on a generic state space simply by calling `logMap()`.

Note that this does not preserve the topology of non-Euclidean state spaces, i.e. the integrator may get "stuck" because it is unable to "wrap around" singularities. This is a fundamental issue until we resolve https://github.com/personalrobotics/aikido/issues/274 - and is true with the current implementation, which simply asks DART to perform the `logMap()` for us.